src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/xresilience/StormLcmSpec.groovy

```diff
@@ -140,7 +140,7 @@ class StormLcmSpec extends HealthCheckSpecification {
         !networkDeployed && wfmManipulator.deployTopology("network")
         srcBlockData && lockKeeper.reviveSwitch(islUnderTest.srcSwitch, srcBlockData)
         dstBlockData && lockKeeper.reviveSwitch(islUnderTest.dstSwitch, dstBlockData)
-        Wrappers.wait(WAIT_OFFSET + discoveryTimeout) {
+        Wrappers.wait(WAIT_OFFSET + discoveryTimeout * 3) {
```

Comment:
a minor thing or maybe not for this PR, but seems like the `WAIT_OFFSET` doesn't do its job well, now there are in fact two offsets. Also, as I understand, the bigger offset is only needed some times, but other times it's fine? Maybe then a retrying mechanism would work better here.

Reply:
In theory, we have several timeouts for different cases, like discoveryTimeout, statsCollectionInterval, etc. 
Also there's 'WAIT_OFFSET', which is some kind of common time gap that we allow kilda to late on. I.e, in this particular case, we expect that after reviving switches new ISL will be discovered not later than in `discoveryTimeout`, but giving it some additional time (offset).
Unfortunately, in most cases we don't have hard specifications, how long would this or that event should be waited for, before we decide that it takes too long. Also, having 'asynchronous world' inside kilda (I mean kafka and storm) doesn't make things easier for us.

So yes, in theory, we should report bug on each event which didn't happen in (`defined in specifications timeout` + WAIT_OFFSET), but in practice, I'm afraid, this will only end up in hundreds of issues, which nobody is going to investigate and fix :\ So we just try to find the amount of time which makes test both stable enough and still be able to discover kilda slownesses.

The highlighted part of the code has a logical mistake, though. We should always wait for only one  `defined in specifications timeout`, but for different offsets. Probably, we need something like `MINOR_WAIT_OFFSET`, `STANDARD_WAIT_OFFSET` and `LONG_WAIT_OFFSET`. 

`Wrappers.wait()` is a retrying mechanism, it is not an explicit waiter.

Reply:
Changed wait time calculation logic (it's a coincidence, that WAIT_OFFSET equals to discovery.timeout at the moment).

Reply:
With retrying I primarily mean that one execution requires to wait let's say 1s, other require 5s, we wait here 8s just to be sure. With retrying we assert something is true and on fail just repeat it a couple of times later and only then fail. So those times when the assertion is true, we don't wait 8s.

This method on 143 is retrying? I thought you need to pass the timeout and the retry timeout into it.

Reply:
Yes, it retries for given amount of time waiting for X seconds between tries (by default we don't  wait, so don't pass the second argument to function.
E.g. in this particular case, we will be waiting for 60 seconds (15 + 3 * 15), evaluating the closure again and again, until it returns true or until 60 seconds have passed.
Logarithmic (or fibonacci's) increase of wait interval between tries is not implemented, though, if you meant the one.